tfhe/src/integer/gpu/list_compression/server_keys.rs

```diff
@@ -173,15 +173,15 @@ impl CudaCompressionKey {
             .sum();
 
         let num_glwes = num_lwes.div_ceil(self.lwe_per_glwe.0);
-        let glwe_ciphertext_size =
-            glwe_ciphertext_size(compressed_glwe_size, compressed_polynomial_size);
+        let glwe_mask_size = glwe_mask_size(
+            compressed_glwe_size.to_glwe_dimension(),
+            compressed_polynomial_size,
+        );
         // The number of u64 (both mask and bodies)
-        // FIXME: have a more precise memory handling, this is too long and should be
-        // num_glwes * glwe_mask_size + num_lwes
-        let uncompressed_len = num_glwes * glwe_ciphertext_size;
+        let uncompressed_len = num_glwes * glwe_mask_size + num_lwes;
         let number_bits_to_pack = uncompressed_len * self.storage_log_modulus.0;
         let compressed_len = number_bits_to_pack.div_ceil(u64::BITS as usize);
-        let mut packed_glwe_list = CudaVec::new(compressed_len, streams, 0);
+        let mut packed_glwe_list = CudaVec::new(num_glwes * compressed_len, streams, 0);
```

Comment:
That change looks weird to me, compressed_len is computed from uncompressed_len, and, uncompressed_len already takes into account the num_glwe